### DISCUSSION TITLE
Create a custom SQL expression with Entity Framework, or how to make EF interpret your Value Types

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#368] [Announcements]
# Create a custom SQL expression with Entity Framework, or how to make EF interpret your Value Types

A deep-dive at the internals of Entity Framework to come up with a solution to make EF interpret your value types, in this particular case a strongly-typed ID (of the StronglyTypedId NuGet Package).

https://timdeschryver.dev/blog/create-a-custom-sql-expression-with-entity-framework-or-how-to-make-ef-interpret-your-value-types

**Comment by daveMC1971** [comment]
Brilliant! and very well explained 👍

**Comment by alabras** [comment]
Thanks you, this solution helped me to compare (with greater than) between two [ulids](https://github.com/Cysharp/Ulid).

**Comment by zwcloud** [comment]
An easier way to solve the problem is to explicitly convert the strong typed ID to string: https://stackoverflow.com/a/68130999/3427520

Just add an implicit or explicit cast operator to its primitive type:

```csharp
public static explicit operator string(PersonId id)
{
    return Value.ToString();
}

var persons = db.Set<Person>().Where(p => ((string)p.Id).Contains("3FE8DBB6"));
```
